core/src/avm1/globals/file_reference_list.rs

```diff
@@ -0,0 +1,141 @@
+use crate::avm1::activation::Activation;
+use crate::avm1::error::Error;
+use crate::avm1::function::FunctionObject;
+use crate::avm1::globals::as_broadcaster::BroadcasterFunctions;
+use crate::avm1::globals::file_reference::FileReferenceObject;
+use crate::avm1::property_decl::{define_properties_on, Declaration};
+use crate::avm1::{ArrayObject, Executable, NativeObject, Object, ScriptObject, TObject, Value};
+use crate::backend::ui::FileFilter;
+use crate::context::GcContext;
+use gc_arena::GcCell;
+
+const PROTO_DECLS: &[Declaration] = declare_properties! {
+    "browse" => method(browse; DONT_ENUM);
+};
+
+const OBJECT_DECLS: &[Declaration] = declare_properties! {};
+
+pub fn browse<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let file_filters = match args.get(0) {
+        Some(Value::Object(array)) => {
+            // Array of filter objects.
+            let length = array.length(activation)?;
+
+            // Empty array is not allowed
+            if length == 0 {
+                return Ok(false.into());
+            }
+
+            let mut results = Vec::with_capacity(length as usize);
+
+            for i in 0..length {
+                if let Value::Object(element) = array.get_element(activation, i) {
+                    let mac_type = if let Ok(val) = element.get("macType", activation) {
+                        Some(val.coerce_to_string(activation)?.to_string())
+                    } else {
+                        None
+                    };
+
+                    let description = element
+                        .get("description", activation)?
+                        .coerce_to_string(activation)?
+                        .to_string();
+
+                    let extensions = element
+                        .get("extension", activation)?
```

Comment:
nit: Variable is named `extensions` even though it stores the property named `extension`. Unlike `description` above.

Reply:
True, however this is actually a list of ';'-separated extensions, unlike description which is a single value. So I think the plural name is more fitting here